internal/store/store.go

```diff
@@ -865,6 +866,12 @@ func (s Store) GetIngressClassParametersV1Alpha1() (*kongv1alpha1.IngressClassPa
 		)}
 	}
 
+	if ingressClass.Spec.Parameters.Scope == nil || ingressClass.Spec.Parameters.Namespace == nil {
+		return nil, ErrNotFound{
+			message: fmt.Sprintf("IngressClass %s should reference namespaced parameters", ingressClass),
+		}
+	}
```

Comment:
Shouldn't we handle both `Scope` options (`Cluster` and `Namespace`)?  As I understand, when:
- `ingressClass.Spec.Parameters.Scope in ("Cluster", nil)`,  it's ok to have `ingressClass.Spec.Parameters.Namespace` nil and it means `IngressClass` scope is cluster-wide,
- `ingressClass.Spec.Parameters.Scope == "Namespace"`, we should validate if `ingressClass.Spec.Parameters.Namespace` is not empty and it means `IngressClass` scope is namespaced.

 Am I missing something? Does our controller support cluster-wide `IngressClass`?

Reply:
Since `IngressClassParamter` is namespaced, I think we could not suppoer cluster wide `IngressClass` in normal way.

Reply:
Seems like we're going to handle that as separate issue?

Reply:
After some reading I think I understand a little bit more:
* `IngressClass` resource is always cluster-wide - ["The IngressClass API itself is always cluster-scoped."](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingressclass-scope) 
* `IngressClass.Spec.Parameters` is a reference to the resource that might be either namespaced or cluster wide. It's specific to the ingress controller. In our case, [`IngressClassParameters` CRD is namespaced](https://github.com/Kong/kubernetes-ingress-controller/blob/08c96459215e86adb297cc595ee67a5f4a4712f5/PROJECT#L57),
* Given above, I conclude it's a valid approach to not accept `IngressClass.Spec.Parameters` referencing cluster-wide resources as we don't support them - `IngressClassParameters` is designed to be namespaced. 

Proposition/question: maybe to avoid future problems with misconfigured `IngressClass.Spec.Parameters` we could add a validation in admission server so that a user is promptly warned?

Reply:
Ah, I missed one more caveat... `IngressClass.Spec.Parameters` has been extended to handle namespaced object references as a result of this upstream issue: https://github.com/kubernetes/enhancements/issues/2365. The feature landed first in Kubernetes 1.22 behind a feature gate, and the gate was removed in 1.25. 

If I understand correctly, that makes our `IngressClassParameters` compatible only with Kubernetes versions >= 1.22 (with feature gate defined) and 1.25 (out of the box). 

That means that if we want to support Kubernetes < 1.22 we will have to introduce another, cluster scoped CRD.

Reply:
I guess an alternative would be to make that lack of support is clear in docs and adapt our tests to run conditionally only for supported k8s versions.

Reply:
It looks like this was enabled by default in 1.22 and made GA in 1.23--they may have kept the gate around but it's been in place for a while per [the feature gate list](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features). Beyond that, [1.22 will EOL in about a month](https://kubernetes.io/releases/#release-v1-22), so I don't think we have good cause to try and support versions further back than that--had we known this for https://github.com/Kong/kubernetes-ingress-controller/pull/2883#issue-1359432365 I would have had more reason to convert it to cluster-scoped, but alas--it's out in the wild now and even though it's only been there for a short while we can't really pull it back.

We need to update the compatibility tables, but shouldn't have a need to support both cluster-scoped and namespaced variants.